### ISSUE TITLE
[HOLD for payment 2023-07-26] [$1000] Settings - Menu items are blocked after clicking on them and returning back

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#23052]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com or open Expensify Desktop / App
2. Login with any account
3. Click Settings 
4. Click any menu item (Preferences or Payments)
5. Click 'Back', and as you're doing that, slide the mouse to the left so that it's no longer over the RHP. If you do this right, you will see that the cursor still has the pointer style.
6. Now hover over any menu item in the settings page

**Note: Try to perform the step from 4-6 as quickly as possible in order to reproduce this bug.**.

## Expected Result:
User can select a menu item and go to it

## Actual Result:
Menu items are blocked

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.42.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/4c64ba11-2aa5-4355-a972-5cf6aca86e64

https://github.com/Expensify/App/assets/78819774/0cf50fd8-e7a2-4a20-9090-187e91866062

https://github.com/Expensify/App/assets/78819774/aec47a14-76a2-493a-81dd-f9fa459d107a

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010e6b7d519b1196ad</li>
        <li>Upwork Job ID: 1681229032377053185</li>
        <li>Last Price Increase: 2023-07-18</li>
    </ul>
</details>

**Comment by techievivek** [comment]
I am unable to reproduce this on my side. @lanitochka17 Can you confirm if this is still happening for you? Thanks.

**Comment by s-alves10** [comment]
This can be reproducible on staging. Please try `Preferences` menu. Click it and back as fast as possible.

**Comment by techievivek** [comment]
Thanks @s-alves10, I can reproduce this now.

**Comment by techievivek** [comment]
Taking off the deployblocker label since this issue is only reproducible when performing the mentioned step very quickly.

**Comment by Moeez905** [comment]
event-listener is malfunctioning

**Comment by Moeez905** [comment]
Contributor details
Your Expensify account email: moeez494216@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0115ba8d118f4a82da

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Settings menu items are blocked after clicking on them and returning back quickly

### What is the root cause of that problem?

We set `isExecuting` state to true before handling `onPress` event and set it back to false once done. Here we use `InteractionManager.runAfterInteractions` to ensure animation was completed before enable the button again.
https://github.com/Expensify/App/blob/be7b6a09cde880cd96470e4e9e0c2f921499dd20/src/components/Pressable/PressableWithFeedback.js#L75-L85

But the callback `InteractionManager.runAfterInteractions` happens not to be called sometimes as described below
https://github.com/facebook/react-native/issues/8624

I verified this by tests. This is the root cause

### What changes do you think we should make in order to solve the problem?

We can try using other scheduling methods such as `setTimeout`.
Below is the custom `runAfterInteractions` function

```
    runAfterInteractions: (callback, timeout = 500) => {
        // ensure callback get called, timeout at 500ms
        // @workaround https://github.com/facebook/react-native/issues/8624
        let called = false;
        const timer = setTimeout(() => {
            called = true;
            callback();
        }, timeout);
        InteractionManager.runAfterInteractions(() => {
            if (called) return;
            clearTimeout(timer);
            callback();
        });
    },
```

Obviously this works as expected

### What alternative solutions did you explore? (Optional)
requestAnimationFrame() can be used as well instead of `setTimeout`

**Comment by mvtglobally** [comment]
@techievivek we are facing this across platforms and not always when you click very very fast. And same is definitely not reproducible in Production

**Comment by techievivek** [comment]
@mvtglobally Looks like this is a major regression and I just experienced this on LHN as well.

**Comment by makiour** [comment]
Hello @techievivek 

I am Mohammed, an experienced full-stack developer and worked as SWE engineer intern at Google and Meta and currently working remotely as an SWE intern at a Fintech company in Dubai.

I am curious if this is still open. I would be more than happy to help with the root cause of this issue and collaborate on the same business day to fix it

Let me know if it is the case!

**Comment by techievivek** [comment]
CC @roryabraham, it looks like this could be happening because of the changes introduced in this PR https://github.com/Expensify/App/pull/21852/files. We tried looking at the changes to see what could be causing the above regression but no dice. Would you be able to take over this and find a fix? Thanks.

**Comment by roryabraham** [comment]
Some thoughts (partially responding to comments in the [slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1689681001193879)):

- Pretty sure 1800 is the correct timeout. This is what caused the "copy to clipboard" icon in the context menu to turn to a checkmark when clicked, then back into the "copy" icon after 1800ms.
- I am a bit skeptical that `useThrottledButtonState` is coming into play here at all because in the `HeaderWithBackButton` component it's only used for the download button, which isn't shown on any of the affected pages.

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/pull/22702 looks like another potential suspect

**Comment by Tushu17** [comment]
@mountiny I reported this bug ([here](https://expensify.slack.com/archives/C049HHMV9SM/p1689636211130979)) before it was logged, So can you check if I'm eligible as a reporter.

**Comment by kbecciv** [comment]
QA team reported first at 6:49 PM EST and @Tushu17 Reported later at 7:23 PM EST
![image](https://github.com/Expensify/App/assets/93399543/25c073c7-51a8-4405-9781-e8d384f96136) 
![image](https://github.com/Expensify/App/assets/93399543/13ddff68-5002-4151-a746-13ce283d2241)

**Comment by roryabraham** [comment]
This happens because PressableWithFeedback gets stuck in state `isExecuting=true` https://github.com/Expensify/App/blob/b019b361da05fed8640311ad0654f15c0ab3c269/src/components/Pressable/PressableWithFeedback.js#L42

cc @robertKozik might be some race condition?

**Comment by roryabraham** [comment]
Maybe this happens when a promise returned from an `onPress` callback never resolves or rejects: https://github.com/Expensify/App/blob/b019b361da05fed8640311ad0654f15c0ab3c269/src/components/Pressable/PressableWithFeedback.js#L82-L84

**Comment by roryabraham** [comment]
Or more likely `InteractionManager.runAfterInteractions` sometimes fails to run

**Comment by roryabraham** [comment]
Maybe related issue: https://github.com/IjzerenHein/react-navigation-shared-element/issues/94

**Comment by robertKozik** [comment]
As far as I remember I wasn't in charge of implementing runAfterInteractions&executing state, I believe this was @priyeshshah11 PR, so maybe he has more thoughts on this.

If the executing state is the source of the problem, maybe we can wrap the `InteractionManager.runAfterInteraction` with some failSafe mechanism which would change this kind of state after some reasonable time?

**Comment by roryabraham** [comment]
Yeah, I've confirmed that at some point _all instances of `InteractionManager.runAfterInteraction`_ just stop running.

**Comment by roryabraham** [comment]
Tried patching our react-native-web fork to include https://github.com/necolas/react-native-web/commit/47d77ac256cf1324e684362465a2b0001d2db574, but still no dice

**Comment by roryabraham** [comment]
So the root cause of this bug is indeed https://github.com/Expensify/App/pull/21852 – the lottie animation on the PreferencesPage is running, and it causes `InteractionManager.runAfterInteractions` to not run. So two possibilities:

- This might mean that the animation isn't properly cleaned up since `requestIdleCallback` isn't called.
- For React Native animations there is an undocumented animation config called `{isInteraction: false}` that you can pass to tell `InteractionManager` to ignore that animation. I'm not sure there's an equivalent for web

**Comment by roryabraham** [comment]
Got a PR up to fix this: https://github.com/Expensify/App/pull/23145

**Comment by jasperhuangg** [comment]
We are CPing a fix for this, I'll retest this once the CP is on staging

**Comment by jasperhuangg** [comment]
Retested on staging and this looks like it isn't happening anymore, closing!

**Comment by kbecciv** [comment]
Issue is not fixed on Desktop environment build is 1.3.42.25

https://github.com/Expensify/App/assets/93399543/d2028f90-406e-4a10-83a4-e7d7f8cd837c

**Comment by roryabraham** [comment]
@kbecciv I'm not able to reproduce this with the Preferences page anymore. Any idea what you did before that happened? Need reproduction steps
